Clean/Circomlib/BinSum.lean

```diff
@@ -0,0 +1,144 @@
+import Clean.Circuit
+import Clean.Utils.Bits
+import Clean.Gadgets.Bits
+
+namespace Circomlib
+open Utils.Bits
+variable {p : ℕ} [Fact p.Prime] [Fact (p > 2)]
+
+-- Define a 2D vector type for BinSum inputs
+-- Represents ops operands, each with n bits
+-- This is a vector of ops elements, where each element is a vector of n field elements
+@[reducible]
+def BinSumInput (n ops : ℕ) := ProvableVector (fields n) ops
+
+-- Instance for NonEmptyProvableType for fields when n > 0
+instance {n : ℕ} [hn : Fact (0 < n)] : NonEmptyProvableType (fields n) where
+  size := n
+  toElements := id
+  fromElements := id
+  nonempty := hn.out
```

Comment:
wouldn't this be enough, since `NonEmptyProvableType` extends `ProvableType`?
```suggestion
instance {n : ℕ} [hn : Fact (0 < n)] : NonEmptyProvableType (fields n) where
  nonempty := hn.out
```

Reply:
another comment, I believe `NeZero n` is the canonical way to encode the non-zero assumptions as a type class